src/dns/aardvark.rs

```diff
@@ -0,0 +1,301 @@
+use crate::network::types;
+use nix::sys::signal::{self, Signal};
+use nix::unistd::Pid;
+use std::collections::HashMap;
+use std::fs;
+use std::fs::File;
+use std::fs::OpenOptions;
+use std::io::prelude::*;
+use std::io::Result;
+use std::net::IpAddr;
+use std::net::Ipv4Addr;
+use std::path::Path;
+use std::process::Command;
+use sysinfo::SystemExt;
+
+const AARDVARK_BINARY: [&str; 1] = ["/usr/libexec/podman/aardvark-dns"];
+
+#[derive(Debug, Clone)]
+pub struct AardvarkEntry {
+    pub network_name: String,
+    pub network_gateway_v4: String,
+    pub network_gateway_v6: String,
+    pub container_id: String,
+    pub container_ip_v4: String,
+    pub container_ip_v6: String,
+    pub container_name: Vec<String>,
+}
+
+#[derive(Debug, Clone)]
+pub struct Aardvark {
+    // aardvark's config directory
+    pub config: String,
+}
+
+impl Aardvark {
+    pub fn new(config: String) -> Self {
+        Aardvark { config }
+    }
+
+    pub fn check_aardvark_support() -> bool {
+        for key in AARDVARK_BINARY {
+            if Path::new(key).exists() {
+                return true;
+            }
+        }
+        log::debug!("No aardvark support found");
+        false
+    }
+
+    // On success retuns aardvark server's pid or returns -1;
+    fn get_aardvark_pid(&mut self) -> i32 {
+        let pid: i32;
+        let path = Path::new(&self.config).join("aardvark.pid".to_string());
+        match fs::read_to_string(&path) {
+            Ok(content) => {
+                pid = match content.parse::<i32>() {
+                    Ok(val) => val,
+                    Err(_) => {
+                        return -1;
+                    }
+                }
+            }
+            Err(_) => {
+                return -1;
+            }
+        }
+
+        pid
+    }
+
+    pub fn start_aardvark_server_if_not_running(&mut self) -> Result<()> {
+        let aardvark_pid = self.get_aardvark_pid();
+        if aardvark_pid != -1 {
+            // check if pid is running
+            let system = sysinfo::System::new_all();
+            match system.process(sysinfo::Pid::from(aardvark_pid)) {
```

Comment:
Can we just send `0` as signal to check if the process is alive? `sysinfo` is a huge dependency that we really do not need just to check for a single process

Reply:
Nice idea `0` is not supported so using `SIGWINCH`